kit/ChildSession.cpp

```diff
@@ -3417,8 +3417,9 @@ void ChildSession::loKitCallback(const int type, const std::string& payload)
             else if (tokens.size() == 3 && tokens.equals(0, "EMPTY"))
             {
                 // with mode:    "EMPTY, <part>, <mode>"
-                const std::string part = (_docType != "text" ? tokens[1].c_str() : "0"); // Writer renders everything as part 0.
-                const std::string mode = (_docType != "text" ? tokens[2].c_str() : "0"); // Writer is not using mode.
+                std::string part = (_docType != "text" ? std::to_string(getLOKitDocument()->getPart()) : "0"); // Writer renders everything as part 0.
+                std::string mode = (_docType != "text" ? std::to_string(getLOKitDocument()->getEditMode()) : "0"); // Writer is not using mode.
+
                 sendTextFrame("invalidatetiles: EMPTY, " + part + ", " + mode +
                               " wid=" + std::to_string(getCurrentWireId()));
```

Comment:
does that mean we ingore what was requested? (part)

Reply:
Callback is sent from the core side. I think we confirm the request's part and mode rather than ignoring :) And this case is relevant when core side sends mode and part numbers with an "EMPTY" string. I think this case occurs mostly when we switch modes in Impress.

The reason for confirming is that while core side sends the callback, the window pointer is empty in Impress. That null pointer causes core side to send "0" instead of the actual part and mode. This happens if user switches from normal view to notes view.

Other than above scenario, getPart and getEditMode should fetch the same values.

Reply:
what if `EMPTY` will be invoked for other page ? eg. we do massive invalidation like changed the page format/size ? isn't that a case here?

Reply:
Now i tested it and this more and it seems this is not a good solution. I need to narrow-down the cases. Moving slides or pages can cause unwanted behaviour. I will re-check a core side solution again or will add a corner case for the notes switch.

Thanks for the comments.